src/network/protocol/messages/lib.zig

```diff
@@ -11,6 +11,24 @@ pub const FeeFilterMessage = @import("feefilter.zig").FeeFilterMessage;
 pub const SendCmpctMessage = @import("sendcmpct.zig").SendCmpctMessage;
 pub const FilterClearMessage = @import("filterclear.zig").FilterClearMessage;
 pub const FilterAddMessage = @import("filteradd.zig").FilterAddMessage;
+pub const NotFoundMessage = @import("notfound.zig").NotFoundMessage;
+
+pub const InventoryVector = struct {
+    type: u32,
+    hash: [32]u8,
+
+    pub fn serializeToWriter(self: InventoryVector, writer: anytype) !void {
+        try writer.writeInt(u32, self.type, .little);
+        try writer.writeAll(&self.hash);
+    }
+
+    pub fn deserializeReader(r: anytype) !InventoryVector {
+        return InventoryVector{
```

Comment:
you forgot the comptime requirement of `anytype` having the writeInt, writeAll, readInt, readBytesNoEof

Reply:
Ohhhh lol, will implement it now